### ISSUE TITLE
Screen Reader handling

### ISSUE THREAD

**Issue by HexaVault** [HexaVault/AD-breaketernity#68]
Stuff to handle screen readers - This could be alt text for achievements or sliders

**Comment by Keithcat1** [comment]
I'm using a screen reader and made a mod that allows me to play the game decidedly more enjoyably:
[https://github.com/Keithcat1/AntimatterDimensionsSourceCode](https://github.com/Keithcat1/AntimatterDimensionsSourceCode)
Instead of making things accessible by default, it has a screen-reader mode which you turn on in options > visual and it presumably breaks some parts of the UI from a sighted perspective, but you can at least se which problems this wants to solve. A few changes are drop-in though, like aria-live on notifications makes achievements nicer and no-one should notice.
I have a horrible hack for making the coding area in the Automator not look like an empty black hole with nothing in it, but looking on their website, the latest version of Codemirror does a significantly better job at accessibility and it might be possible to upgrade to it.

**Comment by HexaVault** [comment]
Yeah i'm not a huge fan of how its an option and i don't really want to just break visuals on a ton of unnecessary stuff just for screen reader, though some of the strategies you use are valid for screen readers (i.e. the upgrades as buttons for perks, though usually we just use hidden upgrade buttons far off screen for that rather than replacing the tree structure)

**Comment by Keithcat1** [comment]
In many cases you can use aria attributes to tell a screen reader what something is and what state it's in without creating another HTML element, so for a checkbox that's written as `<div>` you'd do something like:
```
<div aria-role="checkbox" :aria-checked="checked"></div>
```
If you don't want to just turn them into normal checkboxes. I believe autobuyers has some, time studies tab has one and some more in the import dialog, automator probably has some. If you don't do this there will be no indication to the player weather it's checked, and also allows jumping to it really fast by pressing shortcut keys.
Actually you sort of already have this option, it's called UI. Classic is more accessible because tabs and subtabs are buttons and they're at the top.

**Comment by HexaVault** [comment]
Classic is just an alternate ui, not a screen reader based one. Either way, not every screen reader user is absolutelt blind, some with just have really poor vision, but not none, or some people might just use it as an alternative to wearing glasses (which can hurt after prolonged use, especially if you wear headphones over the top).

We can use aria attributes, sure, but some of the game will be very difficult to navigate anyways and will likely want a set of upgrades that can be tabber through, rather than moving a mouse to try click them, or following some weird order

**Comment by Keithcat1** [comment]
[tabindex](https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Global_attributes/tabindex) works for this.
